firedrake/assemble.py

```diff
@@ -57,7 +57,8 @@ def assemble(expr, tensor=None, bcs=None, *,
              mat_type=None,
              sub_mat_type=None,
              appctx={},
-             options_prefix=None):
+             options_prefix=None,
+             adj=False):
```

Comment:
`assemble` may not want to know if it is assembling `adjoint` or not (I think it should just assemble a given `expr`), but we can think about this after you get things working.

Reply:
I did this so that _collect_lgmaps would know whether it is assembling adjoint or not but I agree that this is a bit hacky